scylla-cql/src/serialize/value.rs

```diff
@@ -855,6 +903,101 @@ fn serialize_sequence<'t, 'b, T: SerializeValue + 't>(
         .map_err(|_| mk_ser_err_named(rust_name, typ, BuiltinSerializationErrorKind::SizeOverflow))
 }
 
+fn serialize_next_constant_length_elem<'t, T: SerializeValue + 't>(
+    rust_name: &'static str,
+    element_type: &ColumnType,
+    typ: &ColumnType,
+    builder: &mut CellValueBuilder,
+    element: &'t T,
+) -> Result<(), SerializationError> {
+    T::serialize(
+        element,
+        element_type,
+        builder.make_sub_writer_without_size(),
+    )
+    .map_err(|err| {
+        mk_ser_err_named(
+            rust_name,
+            typ,
+            VectorSerializationErrorKind::ElementSerializationFailed(err),
+        )
+    })?;
+    Ok(())
+}
+
+fn serialize_next_variable_length_elem<'t, T: SerializeValue + 't>(
+    rust_name: &'static str,
+    element_type: &ColumnType,
+    typ: &ColumnType,
+    builder: &mut CellValueBuilder,
+    element: &'t T,
+) -> Result<(), SerializationError> {
+    let mut element_buffer = Vec::new();
+    let inner_writer = CellWriter::new_without_size(&mut element_buffer);
+    T::serialize(element, element_type, inner_writer).map_err(|err| {
+        mk_ser_err_named(
+            rust_name,
+            typ,
+            VectorSerializationErrorKind::ElementSerializationFailed(err),
+        )
+    })?;
+    let mut element_length_buffer = Vec::new();
+    unsigned_vint_encode(
+        element_buffer.len().try_into().unwrap(),
+        &mut element_length_buffer,
+    );
```

Comment:
It seems a bit wasteful to allocate a Vec just for this. I suppose that vectors with variable length elements won't be too common, so I suppose this is not a huge issue.

Reply:
> It seems a bit wasteful to allocate a Vec just for this. I suppose that vectors with variable length elements won't be too common, so I suppose this is not a huge issue.

Won't they? There are quite a lot of CQL types that are considered variable-length due to god-know-why Cassandra choices...

Reply:
I raised it here before: https://github.com/scylladb/scylla-rust-driver/pull/1165#discussion_r2055690404
As you said, this is most likely not important, and if it is then we can fix it later.

Reply:
> Won't they?

For storing embeddings, I suppose that the default choice would be to use a vector of `float`s or `double`s.

Reply:
I'll mark that as resolved since we won't be addressing this now afaiu